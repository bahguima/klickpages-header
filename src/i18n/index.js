import url from '@/helpers/url';
import Vue from 'vue';
import VueI18n from 'vue-i18n';
import config from './config';
import { getLocale } from './cookie';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('@/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);

    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });

  return messages;
}

const getDefinedLocale = () => {
  const urlLocale = url.getLocale(window.location.pathname);
  const cookieLocale = getLocale();

  return urlLocale || cookieLocale || config.DEFAULT_LOCALE;
};

const i18n = new VueI18n({
  locale: getDefinedLocale(),
  messages: loadLocaleMessages(),
});

export default i18n;
