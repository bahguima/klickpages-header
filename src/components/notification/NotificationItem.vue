<template>
  <li class="drop__item drop__item--notifications">
    <span class="date">{{notification.created_at}}</span>
    <span class="title">{{notification.data[locale].title}}</span>
    <p>{{notification.data[locale].description}}</p>
    <a
      :href="notification.data[locale].link"
      target="_blank">{{notification.data[locale].linkText}}
    </a>
  </li>
</template>

<script>
import { getDefinedLocale } from '../../i18n/index';

export default {
  props: {
    notification: Object,
  },
  computed: {
    locale() {
      const definedLocale = getDefinedLocale();
      return definedLocale === 'pt-BR' ? 'ptBR' : definedLocale;
    },
  },
};
</script>
