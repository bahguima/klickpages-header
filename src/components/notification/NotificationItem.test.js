import { shallowMount, createLocalVue } from '@vue/test-utils';
import { getDefinedLocale } from '@/i18n';

import NotificationItem from './NotificationItem.vue';

jest.mock('@/i18n', () => ({
  t: jest.fn((str) => str),
  getDefinedLocale: jest.fn().mockImplementation(() => 'pt-BR'),
}));

describe('components/notification/NotificationItem', () => {
  const localVue = createLocalVue();
  const notification = {
    id: '1',
    data: {
      ptBR: {
        title: 'Teste',
        description: 'notificacao de teste',
        link: '',
        linkText: '',
      },
      en: {
        title: 'Test',
        description: 'test notification',
        link: '',
        linkText: '',
      },
      es: {
        title: 'Prueba',
        description: 'notificaciÃ³n de prueba',
        link: '',
        linkText: '',
      },
    },
    created_at: '12/05/2020',
  };

  let wrapper;
  const propsData = { notification };

  describe('when component is mounted', () => {
    beforeAll(() => {
      wrapper = shallowMount(NotificationItem, { localVue, propsData });
    });

    afterAll(() => {
      wrapper.destroy();
    });

    it('should renders html correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('computed', () => {
    describe('locale', () => {
      describe('when locale is pt-BR', () => {
        beforeAll(() => {
          wrapper = shallowMount(NotificationItem, { localVue, propsData });
        });

        afterAll(() => {
          wrapper.destroy();
        });

        it('should return ptBR', () => {
          expect(wrapper.vm.locale).toEqual('ptBR');
        });
      });

      describe('when is different from pt-BR', () => {
        const locale = 'en';

        beforeAll(() => {
          getDefinedLocale.mockImplementation(() => locale);
          wrapper = shallowMount(NotificationItem, { localVue, propsData });
        });

        afterAll(() => {
          wrapper.destroy();
        });

        it('should return the defined locale', () => {
          expect(wrapper.vm.locale).toEqual(locale);
        });
      });
    });
  });
});
