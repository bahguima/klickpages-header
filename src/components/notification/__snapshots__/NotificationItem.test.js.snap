// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/notification/NotificationItem when component is mounted should renders html correctly 1`] = `
<li
  class="drop__item drop__item--notifications"
>
  <span
    class="date"
  >
    12/05/2020
  </span>
   
  <span
    class="title"
  >
    Teste
  </span>
   
  <p>
    notificacao de teste
  </p>
   
  <a
    href=""
    target="_blank"
  >
    
  
  </a>
</li>
`;
